components/tidb_query_vec_expr/src/impl_encryption.rs

```diff
@@ -118,6 +120,42 @@ pub fn random_bytes(_ctx: &mut EvalContext, arg: Option<&Int>) -> Result<Option<
     }
 }
 
+pub struct MySQLRng {
+    seed1: u32,
+    seed2: u32,
+}
+
+impl MySQLRng {
+    pub fn new() -> Self {
+        let current_time = time::get_time();
+        let nsec = i64::from(current_time.nsec);
+        Self::new_with_seed(nsec)
+    }
+
+    pub fn new_with_seed(seed: i64) -> Self {
+        let seed1 = (seed.wrapping_mul(0x10001).wrapping_add(55555555)) as u32 % MAX_RAND_VALUE;
+        let seed2 = (seed.wrapping_mul(0x10000001)) as u32 % MAX_RAND_VALUE;
+        MySQLRng { seed1, seed2 }
+    }
+
+    pub fn gen(&mut self) -> f64 {
+        self.seed1 = (self.seed1 * 3 + self.seed2) % MAX_RAND_VALUE;
+        self.seed2 = (self.seed1 + self.seed2 + 33) % MAX_RAND_VALUE;
+        f64::from(self.seed1) / f64::from(MAX_RAND_VALUE)
+    }
+}
```

Comment:
I suggest moving to `impl_math.rs` or a auxiliary `util`/`misc` mod.

Reply:
I prefer a `util` mod.